scilpy/segment/streamlines.py

```diff
@@ -2,56 +2,71 @@
 
 import itertools
 
+from dipy.io.stateful_tractogram import StatefulTractogram
 from dipy.tracking.metrics import length
-from dipy.tracking.streamline import (set_number_of_points,
-                                      transform_streamlines)
-from dipy.tracking._utils import _mapping_to_voxel
+from dipy.tracking.streamline import set_number_of_points
 from dipy.tracking.vox2track import _streamlines_in_mask
 from nibabel.affines import apply_affine
 import numpy as np
 
 
-def target_line_based(streamlines, target_mask, affine=None, include=True):
+def streamlines_in_mask(sft, target_mask):
+    """
+    Parameters
+    ----------
+    sft : StatefulTractogram
+        StatefulTractogram containing the streamlines to segment.
+    target_mask : numpy.ndarray
+        Binary mask in which the streamlines should pass.
+    Returns
+    -------
+    ids : list
+        Ids of the streamlines passing through the mask.
+    """
+    sft.to_vox()
+    sft.to_corner()
     # Copy-Paste from Dipy to get indices
     target_mask = np.array(target_mask, dtype=np.uint8, copy=True)
-    lin_T, offset = _mapping_to_voxel(affine)
 
-    streamline_index = _streamlines_in_mask(
-        streamlines, target_mask, lin_T, offset)
+    streamlines_case = _streamlines_in_mask(list(sft.streamlines),
+                                            target_mask, np.eye(3),
+                                            [-0.5, -0.5, -0.5])
 
-    target_indices = []
-    target_streamlines = []
-    for idx in np.where(streamline_index == [0, 1][include])[0]:
-        target_indices.append(idx)
-        target_streamlines.append(streamlines[idx])
-
-    return target_streamlines, target_indices
+    return np.where(streamlines_case == [0, 1][True])[0].tolist()
 
 
 def filter_grid_roi(sft, mask, filter_type, is_not):
```

Comment:
mask is target_mask in the description

wouldn't "is_exclude" be clearer than "is_not"? (sounds like python's is not). In the main script too.